app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt

```diff
@@ -74,11 +80,26 @@ class ContractDetailActivity : BaseActivity(R.layout.contract_detail_activity) {
             }.attach()
             card.arrow.isInvisible = true
             card.root.transitionName = "contract_card"
+            error.retry.setHapticClickListener {
+                model.loadContract(id)
+            }
         }
 
-        model.data.observe(this) {
-            it.bindTo(binding.card)
-            startPostponedEnterTransition()
+        model.data.observe(this) { result ->
+            binding.apply {
+                if (result.isFailure) {
+                    content.remove()
+                    error.root.apply {
+                        show()
+                        setBackgroundColor(context.colorAttr(R.attr.colorSurface))
+                    }
+                } else {
+                    content.show()
+                    error.root.remove()
+                    result.getOrNull()?.bindTo(binding.card)
+                    startPostponedEnterTransition()
```

Comment:
We might need this row also when an error occurs - if we do not, there will be a large amount of delay upon attempting to navigate to this activity when it errors

Reply:
do you mean like this? 
```model.data.observe(this) { result ->
            binding.apply {
                if (result.isFailure) {
                    content.remove()
                    error.root.apply {
                        show()
                        setBackgroundColor(context.colorAttr(R.attr.colorSurface))
                    }
                } else {
                    content.show()
                    error.root.remove()
                    result.getOrNull()?.bindTo(binding.card)
                }
                startPostponedEnterTransition()
            }
        }
```

Reply:
Yes!